bodhi/models/models.py

```diff
@@ -1649,19 +1649,35 @@ def check_requirements(self, session, settings):
 
         return True, "All checks pass."
 
+    def negative_karma(self):
+        """Checks for negative karma, Returns True if the update has negative karma"""
+        feedback = defaultdict(int)
+        for comment in self.comments:
+            if not comment.anonymous:
+                feedback[comment.user.name] +=  comment.karma
+        for karma in feedback.values():
+            if karma < 0:
+                return True
+
```

Comment:
This function will return True if there is any negative feedback, but it will return None if there is not any negative feedback. If you tack on a final return False at the end here, this function will always return a boolean, rather than alternating return types. Strange bugs can happen with mixed return types sometimes ☺